File: backend/src/main/java/com/github/mahambach/backend/model/WorldMapInvite.java
package com.github.mahambach.backend.model;

import org.springframework.data.annotation.Id;

public record WorldMapInvite(
        @Id
        String id,
        String ownerId,
        String inviteeId,
        String worldMapId) {

        public WorldMapInvite (WorldMapInviteDto worldMapInviteDto) {
                this(null, worldMapInviteDto.ownerId(), worldMapInviteDto.inviteeId(), worldMapInviteDto.worldMapId());
        }
}

File: frontend/src/types/WorldMapInvite.ts
export type WorldMapInvite = {
    id: string;
    ownerId: string;
    inviteeId: string;
    worldMapId: string;
}

export const emptyWorldMapInvite:WorldMapInvite = {
    id: '',
    ownerId: '',
    inviteeId: '',
    worldMapId: ''
}
